### ISSUE TITLE
Access Violation when click Show Console (1.2)

### ISSUE THREAD

**Issue by Benur21** [bruderstein/PythonScript#102]
Hello. I just installed Notepad++ x86 on my computer, and then I installed PythonScript 1.2.

But the first time I click on Show Console, on PythonScript menu, I get the message "Access Violation" with the title "PluginsManager::runPluginCommand Exception". The next times just display an empty console. But after I restart notepad++, when I click Show Console I get the error again.

The same message appears when I click to run any script (startup.py or some sample), but the message title only says "Exception".

Any ideas on what might be wrong? Thank you.

**Comment by chcg** [comment]
How is this related to #100 ? Is the x64 version working for you and just the 32bit x86 version shows this exception messages? I haven't observed this yet.

**Comment by Benur21** [comment]
Yes x64 is fine, x86 is not, and this is not related to #100. Why do you think it is? :/

**Comment by chcg** [comment]
@Benur21  Was just wondering why this should be a x86 only problem. Maybe is related to #21

**Comment by Benur21** [comment]
@chcg I just saw your edit.
The plugin dll is in the Notepad++ installation folder, which you can see in the screenshoot above. I installed Notepad++ in a folder in my desktop. The plugins are going to Notepad++\plugins\.

**Comment by Benur21** [comment]
Is it normal that the console shows up empty and doesn't react to these commands?
![image](https://user-images.githubusercontent.com/29681989/45367178-97d9d900-b5d8-11e8-8000-83f4aae08f88.png)

**Comment by contrail12** [comment]
Did anyone figure out what's going on here? Really missing my PythonScript scripts.

**Comment by Benur21** [comment]
@contrail12 As far as I know, nope. :(

**Comment by sasumner** [comment]
@contrail12 Sure, nobody likes waiting on a fix, but if you are missing your scripts, have you tried downgrading to PS 1.1.0.0 or 1.0.8.0?

**Comment by contrail12** [comment]
OK, I went to downgrade and much to my surprise; I found that I was already running PythonScript 1.0.8.0 and was having the same problem identified in the OP. I then upgraded to PythonScript 1.2.0.0, now everything works, and I'm elated!

Here's some more info about my setup in case my observation will help a real dev fix the OP:
- I'm running Windows 10 version 1709
- I'm running Notepad++ v7.5.8 (32-bit)
- I originally installed an earlier version of NP++ and then upgraded so that I would have Plugin Manager 1.3.5.0 to continue using the original plugin repository instead of the new German-hosted repository, as my company isn't cool with that. So I'm thinking that PythonScript 1.0.8.0 was the most recently published version to the original repository, which is why I was unaware that there was a newer version.

Thanks again for the suggestion @sasumner! Also, **PythonScript makes my job a LOT easier to do, and I supper appreciate all of the dev effort that everyone contributes to make it a reality. Thank you everyone!**

**Comment by chcg** [comment]
@Benur21 You should see something like:

> Python 2.7.15 (v2.7.15:ca079a3ea3, Apr 30 2018, 16:30:26) [MSC v.1500 64 bit (AMD64)]
> Initialisation took 4156ms
> Ready.
> >>> help
> Type help() for interactive help, or help(object) for help about object.
> >>> test
> Traceback (most recent call last):
>   File "<console>", line 1, in <module>
> NameError: name 'test' is not defined

on starting the console window. Seems python27.dll is not found at the right place.

**Comment by Benur21** [comment]
Just saying v1.3 (updated using plugin manager) still has this problem.

**Comment by Benur21** [comment]
Hi, I'm back.
It was still giving error, so I did following:
Tried:
Fully uninstalled notepad++. (checked instalation folder and appdata and ensured they were empty)
Reinstalled notepad++ 7.6.1 with all default options (like Next, Next, Next, Install...)
Installed PythonScript_1.3.0.0.msi. (again Next, Next, Next...)
Opened notepad++.
Clicked Plugins button.
????

![image](https://user-images.githubusercontent.com/29681989/50104614-6a797880-0222-11e9-879c-f9602d92b91d.png)
-----------
Then tried:
Fully uninstalled notepad++. (checked instalation folder and appdata and ensured they were empty)
Reinstalled notepad++ 7.5.9 with default options (like Next, Next, Next, Install...)
Executed PythonScript_1.3.0.0.msi. Clicked Remove because it looked like I didn't uninstall it much correctly.
Executed PythonScript_1.3.0.0.msi second time. (Next, Accept, Next, Next, Install....)
Opened notepad++.
Clicked Plugins button, PythonScript, Show Console.
Yes!

![image](https://user-images.githubusercontent.com/29681989/50105287-2edfae00-0224-11e9-93c3-e9153bf418c6.png)
-----------
Then tried:
Correctly Remove PythonScript via PythonScript_1.3.0.0.msi. (yay didn't forget it!)
Fully uninstalled notepad++. (checked instalation folder and appdata and ensured they were empty)
Reinstalled notepad++ 7.5.9, now only with Don't use %APPDATA%. For some reason it still put some xml files in appdata.
Installed PythonScript_1.3.0.0.msi, all default options.
Show Console is working again!

-----------
Next:
Correctly Remove PythonScript via PythonScript_1.3.0.0.msi.
Fully uninstalled notepad++. (checked instalation folder and appdata and ensured they were empty)
Reinstalled notepad++ 7.5.9, to custom location (same location as the first screenshot in this issue). (everything else is default options)
Installed PythonScript_1.3.0.0.msi, all default options except location which is the same as notepad++.
Opened notepad++.
Raged.

![image](https://user-images.githubusercontent.com/29681989/50106120-bb8b6b80-0226-11e9-80a4-05657719186c.png)
But now we know where's the problem. It doesn't work well on custom install location. The ç and õ in the installation path may be breaking it as well.

**Comment by sasumner** [comment]
Nice trail of debugging!  Wish that all/most could be that thorough...

>The ç and õ in the installation path may be breaking it as well.

Possibly related to #112 ?

**Comment by Benur21** [comment]
Now I reinstalled to \Instalacoes\ instead and the console indeed worked. Yes it seems to be related to #112 .

**Comment by chcg** [comment]
@Benur21 Thanks for tracking down the issue to be a duplicate of 112. Interestingly the number of the emergency call ;-)

**Comment by amibct** [comment]
I was able to activate the plugin as below:

1. Install latest NPP on D: ( A drive other than system drive ) to be able to check "do not use App Data..."
2. Install old Plugin Manager plugin in NPP( the updater folder containing "gpup.exe" must be copied manually then )
3. Install Python Script plugin by using Plugin Manager
4. Open npp\plugins folder and move PythonScript.dll to the folder PythonScript already there.

Now it's fine working

**Comment by andry81** [comment]
> Open npp\plugins folder and move PythonScript.dll to the folder PythonScript already there.

OMG, why is the latest release still suggest broken zip file?
https://github.com/bruderstein/PythonScript/releases/tag/v1.3.0
https://github.com/bruderstein/PythonScript/releases/download/v1.3.0/PythonScript_Full_1.3.0.0.zip

```
<root>
 |
+- plugins
     |
    +- PythonScript
     |
    +- PythonScript.dll
```

I spent hours to figure out that the dll must be placed inside the `<root>/plugins/PythonScript` instead of `<root>/plugins`. Just great. -.-

**Comment by sasumner** [comment]
>OMG, why is the latest release still suggest broken zip file?

Broken zip file?  Sigh.

So Notepad++ comes out with a new required folder structure for plugins _AFTER_ PS 1.3.0.0 is released, and that is somehow Pythonscript's fault?

**Comment by andry81** [comment]
> So Notepad++ comes out with a new required folder structure for plugins _AFTER_ PS 1.3.0.0 is released, and that is somehow Pythonscript's fault?

Yes, because related to this topic (Access Violation).
At least it should be addressed in the file name like `PythonScript_Full_1.3.0.0_before_npp7.6.0.zip` or something like that.

**Comment by sasumner** [comment]
@andry81 Contact @chcg and let him know that you have extra time and are willing to take on such administrative duties as renaming files to keep the world happy.

**Comment by vsrawat** [comment]
When someone is saying something, it is improper to cut him short, 
retort to him, taunt him, mock him, put burden on him, etc. so that he 
is silenced.

This is a help group, and people write here when they have a problem, 
mostly they have not already zeroed in the cause of the problem. They 
are giving vent to their frustration when a working thing suddenly stops 
working and their work suffers.

Thanks.
-- 
Rawat

**Comment by sasumner** [comment]
@vsrawat And if people are going to vent their frustration, they really should have their facts in order before doing so.

**Comment by sasumner** [comment]
@andry81 I can't seem to pull any value out of those 2 sentences.  Gives up...

**Comment by vsrawat** [comment]
he faced a problem and he offered a solution that changing naming system 
of file would make it very clear and people will not face such confusion 
in future.

That seems a good easy workable solution.

It should be implemented, without bothering about other things that he said.

Thanks.
-- 
Rawat

**Comment by daveSkr** [comment]
OK, so this little forum spat seems to have died down, only it still IMPOSSIBLE to install PythonScript in Notepad++ and get it to run; first you need to manually extract files to different locations and when you're done and have double-checked everything, it refuses to run on an "Access violation" error. 

Any idea when, if ever, this will be fixed? Because I really liked using PythonScript in Notepad++ ...

**Comment by chcg** [comment]
See https://github.com/bruderstein/PythonScript/releases/tag/v1.5.2 for a version compatible with the current N++ 7.8 release. Pythonscript will be also available again with the next N++ release from PluginAdmin.

**Comment by Ekopalypse** [comment]
@chcg - is there anything left to do? I assume it can be closed.
